pkgs/applications/editors/vscode/extensions/vscode-utils.nix

```diff
@@ -40,9 +41,14 @@ let
       {
         pname = "vscode-extension-${pname}";
 
-        passthru = passthru // {
-          inherit vscodeExtPublisher vscodeExtName vscodeExtUniqueId;
-        };
+        passthru =
+          {
+            updateScript = vscode-extension-update-script { };
+          }
+          // passthru
```

Comment:
@ShamrockLee Can you have a look at this and let me know if this is the proper way to do this?

Reply:
I'll take a look this wekend.

Reply:
I haven't read through the complete update script. Here are some comments:

1. Update scripts based on `vsce` is certainly a good idea.
2. Current `vscode-utils.fetchVsixFromVscodeMarketplace` uses the "legacy" URL pattern instead of the one . I had a concern that it didn't support `targetPlatform` aside from `universal (https://github.com/NixOS/nixpkgs/pull/161369#issuecomment-1483854812), but such concern is now addressed by PR #178513.